konig-schemagen/src/main/java/io/konig/schemagen/TableShapeGenerator.java

```diff
@@ -0,0 +1,161 @@
+import org.openrdf.model.impl.URIImpl;
+import org.openrdf.model.vocabulary.XMLSchema;
+
+import io.konig.core.KonigException;
+import io.konig.core.NamespaceManager;
+import io.konig.maven.TableShapeGeneratorConfig;
+import io.konig.shacl.NodeKind;
+import io.konig.shacl.PropertyConstraint;
+import io.konig.shacl.Shape;
+import io.konig.shacl.ShapeManager;
+import io.konig.shacl.io.ShapeFileGetter;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.model.FileSet;
+import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.statement.Statement;
+import net.sf.jsqlparser.statement.create.table.ColDataType;
+import net.sf.jsqlparser.statement.create.table.ColumnDefinition;
+import net.sf.jsqlparser.statement.create.table.CreateTable;
+
+public class TableShapeGenerator {
+	
+	private SQLShapeGenerator sqlShapeGenerator = new SQLShapeGenerator();
+
+	private NamespaceManager nsManager;
+	private TableShapeGeneratorConfig tableConfig;
+	
+	public TableShapeGenerator(NamespaceManager nsManager,TableShapeGeneratorConfig tableConfig) {
+		this.nsManager = nsManager;
+		this.tableConfig = tableConfig;
+	}
+
+	public void generateTable(File shapesDir) {
+		ShapeFileGetter fileGetter = new ShapeFileGetter(shapesDir, nsManager);
+		FileSet[] fileSets = tableConfig.getTableFiles();
+		for(FileSet fileSet : fileSets) {
+			if(fileSet.getDirectory() != null){
+				File viewDir = new File(fileSet.getDirectory());
+				File[] files = viewDir.listFiles();
+				for(File file : files) {
+					try (InputStream inputStream = new FileInputStream(file)) {
+						String sqlQuery = IOUtils.toString(inputStream);
+						Statement createTableStmt = CCJSqlParserUtil.parse(sqlQuery); 
+						CreateTable createTable = (CreateTable)createTableStmt;
+						List<ColumnDefinition> columnList = createTable.getColumnDefinitions();
+						Shape shape = createTableShape(createTable,columnList);
+						sqlShapeGenerator.writeShape(shape, fileGetter, nsManager);
+					} catch(Exception ex) {
+						throw new KonigException(ex);
+					}
+				}
+			}	
+		}
+	}
+	
+	
+	private Shape createTableShape(CreateTable createTable, List<ColumnDefinition> columnList) {
+		String URI ="http://example.com/shapes/";
+		String tableName = createTable.getTable().getName();
+		String shapeId = URI+tableName;
+		Shape shape = new Shape(new URIImpl(shapeId));
+		addPropertyContraint(shape, columnList);
+		return shape;
+	}
+		
+	private void addPropertyContraint(Shape shape,  List<ColumnDefinition> columnList) {
+		
+		List<PropertyConstraint> propertyConstraints = new ArrayList<PropertyConstraint>();
+			for(ColumnDefinition column : columnList) {
+				PropertyConstraint pc = new PropertyConstraint();			
+						pc.setPath(new URIImpl(tableConfig.getPropertyNamespace()+column.getColumnName()));
+						pc.setMaxCount(1);
+						pc.setMinCount(getMinCount(column));
+						pc.setFormula(null);
```

Comment:
You need to set the following constraints as appropriate:

- maxExclusive
- minExclusive
- maxInclusive
- minInclusive
- maxLength
- minLength
- decimalPrecision
- decimalScale